app/code/community/Bolt/Boltpay/controllers/OnepageController.php

```diff
@@ -27,69 +27,15 @@ class Bolt_Boltpay_OnepageController
-                if (!$result) {
-                    if (Mage::getStoreConfig('payment/boltpay/skip_payment')) {
-                        try {
-                            $data = array('method' => Bolt_Boltpay_Model_Payment::METHOD_CODE);
-                            $result = $this->getOnepage()->savePayment($data);
-                        } catch (Mage_Payment_Exception $e) {
-                            if ($e->getFields()) {
-                                $result['fields'] = $e->getFields();
-                            }
-
-                            $result['error'] = $e->getMessage();
-                            Mage::helper('boltpay/bugsnag')->notifyException($e);
-                        } catch (Mage_Core_Exception $e) {
-                            $result['error'] = $e->getMessage();
-                            Mage::helper('boltpay/bugsnag')->notifyException($e);
-                        } catch (Exception $e) {
-                            //Mage::logException($e);
-                            $result['error'] = $this->__('Unable to set Payment Method.');
-                            Mage::helper('boltpay/bugsnag')->notifyException($e);
-                        }
-                    }
-
-                    if (!$result) {
-                        Mage::dispatchEvent(
-                            'checkout_controller_onepage_save_shipping_method',
-                            array(
-                                'request' => $this->getRequest(),
-                            'quote' => $this->getOnepage()->getQuote())
-                        );
-                        Mage::helper('boltpay')->collectTotals($this->getOnepage()->getQuote());
-                        $this->getResponse()->setBody(Mage::helper('core')->jsonEncode($result));
-                        if (Mage::getStoreConfig('payment/boltpay/skip_payment')) {
-                            $this->loadLayout('checkout_onepage_review');
-                            $result['goto_section'] = 'review';
-                            $result['update_section'] = array(
-                                'name' => 'review',
-                                'html' => $this->_getReviewHtml());
-                        } else {
-                            $result['goto_section'] = 'payment';
-                            $result['update_section'] = array(
-                                'name' => 'payment-method',
-                                'html' => $this->_getPaymentMethodsHtml());
-                        }
-                    }
-                }
-
-                Mage::helper('boltpay')->collectTotals($this->getOnepage()->getQuote())->save();
-                $this->getResponse()->setBody(Mage::helper('core')->jsonEncode($result));
-            }
-        } catch (Exception $e) {
-            Mage::helper('boltpay/bugsnag')->notifyException($e);
-            throw $e;
-        }
+        $this->willReturnJson = false;
```

Comment:
this class doesn't use ApiTrait. Why these flags need to be set?

Reply:
@daisy1754 Remember, we are doing micro pull request. There is no perfect way to break down so many interdependent changes. Trying entire files will have too many changes, so we will focus in on ideas. The focus here is API Trait capabilities.

This controller/file will rely on api trait capabilities in a later Pull Request. Specifically, this is the controller that creates the success page that takes a Bolt payload and Bolt signature as parameters. By default, however, we want it to behave like a standard controller... not necessarily returning JSON and definitely not requiring a signature.

For now, it is just important to understand that it has been enabled to behave with API controller traits and that in the 1.x context, and success page context, we do not want to declare this as returning JSON.

Reply:
The use statement was accidently omitted in the "cherry-picking" like approach to making these pull request.  I corrected the omission.